### PR TITLE
Generate code coverage output with coverlet

### PR DESCRIPTION
**Pull request by StanleyGoldman**
- [x] Integrate a tool to generate code coverage
- [x] Integrate codecov.io
- [x] Added badge to readme

### CODE REVIEW

**Comment by StanleyGoldman** [comment]
So far I'm having issues using `coverlet` to generate coverage. Opened an issue with them: https://github.com/tonerdo/coverlet/issues/188

**Comment by ryangribble** [comment]
Hey @StanleyGoldman this is awesome! I've been wanting to get coverage stuff in place. I'm not sure about coverlet, I was planning on following what John Skeet has on https://github.com/nodatime/nodatime (using dotcover I believe)

**Comment by StanleyGoldman** [comment]
Hey @ryangribble. Yea, I've just been on a code coverage spree lately.

I got coverlet and codecov working with https://github.com/octokit/octokit.graphql.net and https://github.com/justaprogrammer/MSBuildLogOctokitChecker
I'm using OpenCover and codecov for https://github.com/github/VisualStudio

I'm just having trouble with coverlet and this project, so hopefully they can help me out.

**Comment by StanleyGoldman** [comment]
Also, wouldn't you have to get a license of dotcover for this?

**Comment by ryangribble** [comment]
Hmm I had a feeling you could get dotcover for free on open source projects

**Comment by StanleyGoldman** [comment]
That is quite possible..

**Comment by shiftkey** [comment]
> That is quite possible..

Here, read: https://www.jetbrains.com/buy/opensource/

**Comment by StanleyGoldman** [comment]
Okay. Well I don't know how to feel about code coverage tools. I can't find a comparison of dotCover vs other similar tools, but I did manage to get coverlet operational thanks to help from some of their users. We can use another tool in it's place as they all support a common output format.

Now i'm struggling with codecov.io. I have it integrated but I can't seem to get their report uploader to work correctly.

**Comment by StanleyGoldman** [comment]
@ryangribble I logged some issues for remaining work, but I think this is good to merge.

**Comment by tonerdo** [comment]
Just checking if you're encountering a problem with coverlet that's preventing this from getting merged

**Comment by ryangribble** [comment]
Sadly I let this one drift and I'm struggling to get back up to speed on it 😳 

Apart from the obvious merge conflicts, it seems like some other things could possibly need looking at now due to the passage of time... for example this is using a local implementation that runs coverlet.exe as a cake tool directly, whereas the linked issue indicates this may now be merged into the `Cake.Coverlet` library.

Im also wanting to understand the implications of changing pdb type to portable, and can spot a few things (like hardcoded paths to coverlet.exe) that should be tidied up (declare coverlet path in a more central place, etc)

**Comment by StanleyGoldman** [comment]
@ryangribble I've also been MIA on this, so it's not just you.
I'll update the PR and dependencies used.

And yea, we'll have to find out some more information regarding PDB types.

File: Octokit.Reactive/Octokit.Reactive.csproj [hunk 1]
```diff
@@ -12,7 +12,7 @@
     <NetStandardImplicitPackageVersion>1.6.0</NetStandardImplicitPackageVersion>
     <AssemblyName>Octokit.Reactive</AssemblyName>
     <PackageId>Octokit.Reactive</PackageId>
-    <DebugType>embedded</DebugType>
+    <DebugType>portable</DebugType>
```
[the remark this answers is not in the document]
Comment (StanleyGoldman) [reply Octokit.Reactive/Octokit.Reactive.csproj]: I'm not going to know for sure, but it seems like I may have been working around an issue in coverlet at the time. https://github.com/tonerdo/coverlet/issues/476

I'm testing now with the changed removed, so far it seems like the issue has been resolved.

[the remark this answers is not in the document]
Comment (StanleyGoldman) [reply Octokit.Reactive/Octokit.Reactive.csproj]: Yea, I think it's not needed anymore.

**Comment by shiftkey** [comment]
@StanleyGoldman great - thanks for following up with that helpful context!

**Comment by shiftkey** [comment]
release_notes: Added code coverage reporting to CI build

**Pull request merged**